2021-Firmware-Bootcamp/Core/Src/main.c

```diff
@@ -0,0 +1,217 @@
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "spi.h"
+#include "tim.h"
+#include "usart.h"
+#include "gpio.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+/* USER CODE END PV */
+
+/* Private function prototypes -----------------------------------------------*/
+void SystemClock_Config(void);
+/* USER CODE BEGIN PFP */
+
+/* USER CODE END PFP */
+
+/* Private user code ---------------------------------------------------------*/
+/* USER CODE BEGIN 0 */
+
+/* USER CODE END 0 */
+
+/**
+  * @brief  The application entry point.
+  * @retval int
+  */
+int main(void)
+{
+  /* USER CODE BEGIN 1 */
+
+  /* USER CODE END 1 */
+
+  /* MCU Configuration--------------------------------------------------------*/
+
+  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+  HAL_Init();
+
+  /* USER CODE BEGIN Init */
+
+  /* USER CODE END Init */
+
+  /* Configure the system clock */
+  SystemClock_Config();
+
+  /* USER CODE BEGIN SysInit */
+
+  /* USER CODE END SysInit */
+
+  /* Initialize all configured peripherals */
+  MX_GPIO_Init();
+  MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
+  /* USER CODE BEGIN 2 */
+
+
+
+  //turns gpio high and stop reading from adc
+  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8,GPIO_PIN_SET);
+  uint8_t txData[3] = {1,128,255};
+  uint8_t rxData[3] = {};
+  uint16_t count;
+  HAL_StatusTypeDef hal_status;
+
+  //start pwm output
+  HAL_TIM_PWM_Start(&htim1,TIM_CHANNEL_1);
+
+  const double ADC_CONVERT = 3.128;
+  const int FIVE_PERCENT_COUNT = 3200;
+
+
+
+
+
+
+  /* USER CODE END 2 */
+
+  /* Infinite loop */
+  /* USER CODE BEGIN WHILE */
+  while (1)
+  {
+
+	  //turns gpio low, start reading from adc
+	 	  HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8,GPIO_PIN_RESET);
+	 	  //wait what does spi_timeout do
+	 	  hal_status = HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 3, SPI_TIMEOUT);  //using spi1, transmitting 3 groups of 8 bits
+	 	  if(hal_status != HAL_OK)
+	 	  {
+	 		  break;
+	 	  }
```

Comment:
What is this for here? Breaking out of the while(1) loop? Idk if that would be the greatest idea because in terms of a real embedded just killing your program like this doesn't usually happen for multiple reasons but in this case, it is possible that the hal_status is bad for like a cycle or two but it is able to recover for some reason.